src/home/BudgetAtAGlanceWidget.js

```diff
@@ -1,7 +1,245 @@
+ * Informs user if amount spent is "Under", "On Track", or "Over" budget
+ * Spent percentage is compared to month percentage, status separated by 10%
+ * Displays progress bar
+ *
+ * @component
+ */
+
+function BudgetAtAGlance() {
+  const date = new Date('');
+
+  const firstDay = new Date(date.getFullYear(), date.getMonth(), 1);
+
+  const lastDay = new Date(date.getFullYear(), date.getMonth() + 1, 0);
+
+  // Get all transactions that do not fall under income category and reduce
+  const amountSpent = useSelector(
+    (state) => {
+      const allTransactions = state.transactions.entities
+        .filter(
+          (transaction) => transaction.date >= firstDay && transaction.date <= lastDay,
+        )
+        .filter(
+          (transaction) => transaction.categoryId !== INCOME_ID,
+        )
+        .reduce(
+          (total, transaction) => total + transaction.amount,
+          0,
+        );
+      return parseFloat(allTransactions);
+    },
+  );
+
+  // Gets categories data, excludes income category, sums target values
+  const totalBudget = useSelector(
+    (state) => {
+      const allCategories = state.categories.entities
+        .filter(
+          (category) => category.id !== INCOME_ID,
+        )
+        .reduce(
+          (total, category) => total + category.target,
+          0,
+        );
+      return parseFloat(allCategories);
+    },
+  );
+
+  // Gets budget percentage to calculate status
+  const spentToBudget = parseFloat(Math.abs(amountSpent) / totalBudget);
+
+  // Gets month percentage to calculate status
+  const currDay = date.getDate();
+  const endDay = lastDay.getDate();
+  const monthPercentage = parseFloat(currDay / endDay);
+
+  let status = '';
+  if (amountSpent >= 0 && spentToBudget <= (monthPercentage - 0.10)) {
+    status = 'Under';
+  } else if (amountSpent >= 0 && spentToBudget > (monthPercentage - 0.10)) {
+    status = 'On Track';
+  } else {
+    if (spentToBudget >= (monthPercentage + 0.10)) {
+      status = 'Over';
+    } else if (spentToBudget <= (monthPercentage - 0.10)) {
+      status = 'Under';
+    } else {
+      status = 'On Track';
+    }
+  }
+
+  return (
+    <div className="row">
+      <div className="col-md-6">
+        <div className="card border-secondary mb-3">
+          <h4 className="card-header">Budget at a Glance</h4>
+          <p className="m-3">
+            <SpentTracker amount={amountSpent} total={totalBudget} />
+            <div className="my-3">
+              <BudgetProgressBar
+                amount={amountSpent}
+                total={totalBudget}
+                ratio={spentToBudget}
+                monthPercent={monthPercentage}
+                status={status}
+              />
+            </div>
+            <div className="my-3">
+              <BudgetStatus
+                status={status}
+              />
+            </div>
+          </p>
```

Comment:
Paragraphs shouldn't have block level elements inside of them. I would change this to a div. Also, it looks like it should have the "card-body" class instead of "m-3" to fit the Bootstrap card setup.